### PR TITLE
Ensuring that flush() will raise on rollback if called from change/0.

### PR DESCRIPTION
**Pull request by Eiji7**
ping @josevalim

### CODE REVIEW

File: lib/ecto/migration.ex [hunk 1]
```diff
@@ -1096,14 +1096,16 @@ defmodule Ecto.Migration do
...
-  Executes queue migration commands.
-
-  Reverses the order in which commands are executed when doing a rollback
-  on a `change/0` function and resets the commands queue.
-  """
-  def flush do
-    Runner.flush
+  @doc "Executes queue migration commands."
+  defmacro flush do
+    quote bind_quoted: [caller: __CALLER__.module] do
```
Comment (josevalim) [inline lib/ecto/migration.ex]: You don't need `bind_quoted`, the module inside the quote will be `__MODULE__`.

File: lib/ecto/migration.ex [hunk 2]
```diff
@@ -1096,14 +1096,16 @@ defmodule Ecto.Migration do
...
+      if direction() == :down and function_exported?(caller, :change, 0) do
```
Comment (josevalim) [inline lib/ecto/migration.ex]: The migrator checks for down first and then falls back to `:change`, so we should express this here:

```suggestion
      if direction() == :down and not function_exported?(caller, :down, 0) do
```

File: lib/ecto/migration.ex [hunk 3]
```diff
@@ -1096,14 +1096,16 @@ defmodule Ecto.Migration do
...
+        message = "Calling flush() inside change when doing rollback is not supported."
```
Comment (josevalim) [inline lib/ecto/migration.ex]: Error messages start with lowercase:

```suggestion
        message = "calling flush() inside change when doing rollback is not supported."
```

File: lib/ecto/migration.ex [hunk 4]
```diff
@@ -1096,14 +1096,16 @@ defmodule Ecto.Migration do
...
+        raise ArgumentError, message
```
Comment (josevalim) [inline lib/ecto/migration.ex]: Not an ArgumentError in this case, so this is enough:

```suggestion
        raise message
```

**Comment by josevalim** [comment]
Thanks @Eiji7, I have dropped some comments. Also, instead of adding an integration test, can you please add a unit test? See test/ecto/migrator_test.exs.

**Comment by Eiji7** [comment]
@josevalim Thanks, everything makes sense for me. I have send all suggested changes as well as moved test.

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**